airbyte-db/src/main/java/io/airbyte/db/DatabaseHelper.java

```diff
@@ -33,39 +33,46 @@
 
 public class DatabaseHelper {
 
-  public static BasicDataSource getConnectionPool(String username,
-                                                  String password,
-                                                  String jdbcConnectionString) {
+  // default to postgres (which is the driver that is installed with this library).
+  public static BasicDataSource getConnectionPool(String username, String password, String jdbcConnectionString) {
 
-    BasicDataSource connectionPool = new BasicDataSource();
-    connectionPool.setDriverClassName("org.postgresql.Driver");
+    return getConnectionPool(username, password, jdbcConnectionString, "org.postgresql.Driver");
+  }
+
+  public static BasicDataSource getConnectionPool(String username, String password, String jdbcConnectionString, String driverClassName) {
+    final BasicDataSource connectionPool = new BasicDataSource();
+    connectionPool.setDriverClassName(driverClassName);
     connectionPool.setUsername(username);
     connectionPool.setPassword(password);
     connectionPool.setUrl(jdbcConnectionString);
 
     return connectionPool;
   }
 
+  // default to postgres dialect
   public static <T> T query(BasicDataSource connectionPool, ContextQueryFunction<T> transform) throws SQLException {
+    return query(connectionPool, transform, SQLDialect.POSTGRES);
```

Comment:
An idea here could be to have an object that hides BDS and contains the dialect enum. Once you have that you can reuse this function for all new type of connections